dev/AppNotifications/AppNotificationBuilder/AppNotificationComboBox.cpp

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "AppNotificationComboBox.h"
+#include "Microsoft.Windows.AppNotifications.Builder.AppNotificationComboBox.g.cpp"
+#include "AppNotificationBuilderUtility.h"
+
+namespace winrt::Microsoft::Windows::AppNotifications::Builder::implementation
+{
+    AppNotificationComboBox::AppNotificationComboBox(hstring const& id) : m_id(id) { };
+
+    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationComboBox AppNotificationComboBox::AddItem(winrt::hstring const& id, winrt::hstring const& content)
+    {
+        THROW_HR_IF_MSG(E_INVALIDARG, m_items.Size() >= c_maxSelectionElements, "Maximum number of items added");
+        THROW_HR_IF_MSG(E_INVALIDARG, id.empty(), "You must provide an id when adding an argument");
+
+        m_items.Insert(id, content);
+
+        return *this;
+    }
+
+    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationComboBox AppNotificationComboBox::SetTitle(winrt::hstring const& value)
+    {
+        m_title = value;
+
+        return *this;
+    }
+
+    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationComboBox AppNotificationComboBox::SetSelectedItem(winrt::hstring const& id)
+    {
+        m_selectedItem = id;
+
+        return *this;
+    }
+
+    std::wstring AppNotificationComboBox::GetSelectionItems()
+    {
+        std::wstring items{};
+
+        for (auto pair : m_items)
+        {
+            items.append(wil::str_printf<std::wstring>(L"<selection id='%ls' content='%ls'/>", pair.Key().c_str(), pair.Value().c_str()));
+        }
+
+        return items;
+    }
+
+    winrt::hstring AppNotificationComboBox::ToString()
+    {
+        std::wstring xmlResult{ wil::str_printf<std::wstring>(L"<input id='%ls' type='selection'%ls%ls>%ls</input>",
+            m_id.c_str(),
+            m_title.empty() ? L"" : wil::str_printf<std::wstring>(L" title='%ls'", m_title.c_str()).c_str(),
+            m_selectedItem.empty() ? L"" : wil::str_printf<std::wstring>(L" defaultInput='%ls'", m_selectedItem.c_str()).c_str(),
```

Comment:
We have an opportunity here to do some extra validation that goes beyond what the xml schema prescribes. For example, we could ensure that the value for SelectedITem refers an existing selection. That's probably something worth exploring in a future iteration.

Reply:
Captured here: https://microsoft.visualstudio.com/OS/_workitems/edit/40746502